var/spack/repos/builtin/packages/suite-sparse/package.py

```diff
@@ -271,30 +314,46 @@ def fix_darwin_install(self):
         if "+pic platform=darwin" in self.spec:
             fix_darwin_install_name(self.spec.prefix.lib)
 
-    @property
-    def libs(self):
-        """Export the libraries of SuiteSparse.
-        Sample usage: spec['suite-sparse'].libs.ld_flags
-                      spec['suite-sparse:klu,btf'].libs.ld_flags
-        """
-        # Component libraries, ordered by dependency. Any missing components?
-        all_comps = [
-            "klu",
-            "btf",
-            "umfpack",
-            "cholmod",
-            "colamd",
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    def cmake_args(self):
+        projects = [
+            "suitesparse_config",
+            "mongoose",
             "amd",
+            "btf",
             "camd",
             "ccolamd",
+            "colamd",
+            "cholmod",
             "cxsparse",
             "ldl",
+            "klu",
+            "umfpack",
+            "paru",
             "rbio",
             "spqr",
-            "suitesparseconfig",
+            "spex",
         ]
-        query_parameters = self.spec.last_query.extra_parameters
-        comps = all_comps if not query_parameters else query_parameters
-        return find_libraries(
-            ["lib" + c for c in comps], root=self.prefix.lib, shared=True, recursive=False
-        )
+        if "+graphblas" in self.spec:
+            projects.extend(["graphblas", "lagraph"])
+
+        blas = self.spec["blas"]
+        ilp64 = blas.satisfies("+ilp64")
+
+        args = [
+            self.define("SUITESPARSE_USE_STRICT", True),
+            self.define_from_variant("SUITESPARSE_USE_CUDA", "cuda"),
+            self.define_from_variant("SUITESPARSE_USE_OPENMP", "openmp"),
+            self.define("SUITESPARSE_ENABLE_PROJECTS", projects),
+            self.define("SUITESPARSE_USE_64BIT_BLAS", ilp64),
+            self.define("BLAS_LIBRARIES", blas.libs),
+            self.define("LAPACK_LIBRARIES", self.spec["lapack"].libs),
+        ]
+
+        if blas.name == "openblas" and "symbol_suffix" in blas.variants:
```

Comment:
Why only for OpenBLAS and not for any BLAS with a symbol suffix?
The "default" symbol suffix in SuiteSparse is no suffix at all.